Sources/Realtime/Channel.swift

```diff
@@ -694,13 +694,61 @@ extension Channel {
     return state == .leaving
   }
 }
+// ----------------------------------------------------------------------
+
+// MARK: - Codable Payload
+
+// ----------------------------------------------------------------------
+
+extension Payload {
+    
+    /// Initializes a payload from a given value
+    /// - parameter value: The value to encode
+    /// - parameter encoder: The encoder to use to encode the payload
+    /// - throws: Throws an error if the payload cannot be encoded
+    init<T: Encodable>(_ value: T, encoder: JSONEncoder = Defaults.encoder) throws {
+        let data = try encoder.encode(value)
+        self = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as! Payload
+    }
+    
+    /// Decodes the payload to a given type
+    /// - parameter type: The type to decode to
+    /// - parameter decoder: The decoder to use to decode the payload
+    /// - returns: The decoded payload
+    /// - throws: Throws an error if the payload cannot be decoded
+    public func decode<T: Decodable>(to type: T.Type, decoder: JSONDecoder = Defaults.decoder) throws -> T {
```

Comment:
You can define a default value for the type and eliminate the method overload.

```suggestion
    public func decode<T: Decodable>(to type: T.Type = T.self, decoder: JSONDecoder = Defaults.decoder) throws -> T {
```

Reply:
Replaced all unnecessary decode functions using this syntax instead in 738def4 - thanks for your help @grsouza